src/h5web/utils.ts

```diff
@@ -1,8 +1,13 @@
 import { format } from 'd3-format';
-import type { Entity, Group } from './providers/models';
+import type { Entity, Group, H5WebComplex } from './providers/models';
 
 export const formatValue = format('.3~e');
 export const formatPreciseValue = format('.5~e');
+export function formatComplexValue(value: H5WebComplex, specifier = '') {
+  const [real, imag] = value;
+  const formatFn = format(specifier);
+  return `${formatFn(real)}${imag >= 0 ? '+' : ''}${formatFn(imag)}i`;
```

Comment:
The operator in the middle needs spaces on either side. One option would be add the minus sign in the ternary (with spaces around) and take the absolute value of `imag` in the third block.

Also I would use `Math.sign` in the ternary. What happens if `imag` is equal to 0? I'd argue that `0i` should _not_ be rendered, but maybe that's not what other libraries do 🤷.